gix-fs/src/lib.rs

```diff
@@ -65,9 +65,9 @@ pub fn current_dir(precompose_unicode: bool) -> std::io::Result<PathBuf> {
 #[derive(Clone)]
 pub struct Stack {
     /// The prefix/root for all paths we handle.
-    root: PathBuf,
-    /// the most recent known cached that we know is valid.
-    current: PathBuf,
```

Comment:
These have to stay `PathBuf` as the type is a link between relative paths and the filesystem location they represent.
It's also in the interest of performance to keep it close to what it ought to represent.